operator/v1/types_dns.go

```diff
@@ -162,6 +172,40 @@ type ForwardPlugin struct {
 	Policy ForwardingPolicy `json:"policy,omitempty"`
 }
 
+// UpstreamResolvers defines a schema for configuring the CoreDNS forward plugin in the
+// specific case of the default (".") server.
+// It defers from ForwardPlugin in the default values it accepts:
+// * At least one upstream should be specified.
+// * the default policy is Sequential
+type UpstreamResolvers struct {
+	// upstreams is a list of resolvers to forward name queries for the . domain.
+	// Each instance of CoreDNS performs health checking of Upstreams. When a healthy upstream
+	// returns an error during the exchange, another resolver is tried from Upstreams. The
+	// Upstreams are selected in the order specified in Policy. Each upstream is represented
+	// by an IP address or IP:port if the upstream listens on a port other than 53.
+	//
+	// At least 1 upstream should be specified
+	// A maximum of 15 upstreams is allowed per ForwardPlugin.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=15
+	// +kubebuilder:default:={"/etc/resolv.conf"}
```

Comment:
I'm concerned that this default value implies that multiple files may be passed to the `Upstream` array.

Reply:
Added extra logic in `controller_dns_configmap` in order to prevent a user from mixing /etc/resolv.conf with one or more other resolvers.

Reply:
Let's restrict each entry to either an IPv4 or IPv6 address or the literal string "/etc/resolv.conf".  This cannot be specified using kubebuilder tags, so you'll need to use a patch like we do for the ingresscontroller CRD:  https://github.com/openshift/api/blob/master/operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml-patch

I think you'll need something like this (I haven't test it though):

Reply:
Is it actually invalid to mix `/etc/resolv.conf` in a list of IP addresses?  I haven't tried, but from a quick look at the documentation and source code, it looks like it should work.

Reply:
Looked at the source code and it looks like it should work as well. But should we simply allow a user to specify any valid filepath. Because it would get rendered by the operator and only fail in CoreDNS. Restricting it to only `/etc/resolv.conf` seems like a sensible precaution.

Reply:
@Miciah is there a way we can make this validate according to to forward plugin validation?

> TO... are the destination endpoints to forward to. The TO syntax allows you to specify a protocol, tls://9.9.9.9 or dns:// (or no protocol) for plain DNS. The number of upstreams is limited to 15.

plain IP is not enough, it might need a protocol and port as well.

https://github.com/coredns/coredns/tree/master/plugin/forward#syntax

Reply:
> @Miciah is there a way we can make this validate according to to forward plugin validation?
> 
> > TO... are the destination endpoints to forward to. The TO syntax allows you to specify a protocol, tls://9.9.9.9 or dns:// (or no protocol) for plain DNS. The number of upstreams is limited to 15.
> 
> plain IP is not enough, it might need a protocol and port as well.
> 
> https://github.com/coredns/coredns/tree/master/plugin/forward#syntax

Agreed, we also saw with @arjunrn that a domain name could be used as well, with the resolution of that name in /etc/hosts